plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/alarms/AlarmManager.java

```diff
@@ -0,0 +1,263 @@
+package eu.arrowhead.core.plantdescriptionengine.alarms;
+
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_monitor.dto.PdeAlarmDto;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AlarmManager {
+
+    private final List<Alarm> activeAlarms = new ArrayList<>();
+    private final List<Alarm> clearedAlarms = new ArrayList<>();
+
+    /**
+     * @param id The ID of a PDE Alarm.
+     * @return Data describing the alarm with the given ID if it exists, null
+     * otherwise.
+     */
+    private Alarm getAlarmData(final int id) {
+        for (final Alarm alarm : activeAlarms) {
+            if (alarm.getId() == id) {
+                return alarm;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @return A list containing all PDE alarms.
+     */
+    public List<PdeAlarmDto> getAlarms() {
+        final List<PdeAlarmDto> result = new ArrayList<>();
+
+        final List<Alarm> allAlarms = new ArrayList<>();
```

Comment:
Maybe add activeAlarms.size() + clearedAlarms.size() as constructor parameters. This way you can avoid the ArrayList internal resizing.

Reply:
Fixed